### PR TITLE
[Data Object Editor] Add Endpoint to get fieldcollection

### LINKED ISSUE #575: [Data Object Editor] Add Endpoint to get fieldcollection
`/admin/class/fieldcollection-tree?_dc=1732263392203&allowedTypes=NewsCars%2CNewsLinks%2CNewsText&object_id=773&field_name=content&layoutId=0&forObjectEditor=1`(we only need the forObjectEdit relevant infos without custom layouts for now)

Part of https://github.com/pimcore/studio-backend-bundle/issues/566

### PR DESCRIPTION
**Pull request by martineiber**
## Changes in this pull request
Resolves #575 

## Additional info

Part of https://github.com/pimcore/studio-backend-bundle/issues/566

### CODE REVIEW

File: src/Class/Hydrator/LayoutDefinitionHydrator.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+use Pimcore\Bundle\StudioBackendBundle\Class\Schema\LayoutDefinition;
+use Pimcore\Model\DataObject\Fieldcollection\Definition as FieldCollectionDefinition;
+
+/**
+ * @internal
+ */
+final class LayoutDefinitionHydrator implements LayoutDefinitionHydratorInterface
+{
+    public function hydrate(FieldCollectionDefinition $data): LayoutDefinition
```
Comment (lukmzig) [inline src/Class/Hydrator/LayoutDefinitionHydrator.php]: Should we maybe make this a little more specific? Either use `Field Collection namespace` or if we will share the hydrator, then rename the method to `hydrateFieldCollection` or similiar? WDYT?

Comment (martineiber) [reply src/Class/Hydrator/LayoutDefinitionHydrator.php]: I will but it into a Field Collection namespace like all the other Service. 

Good Point thx :)

Comment (martineiber) [reply src/Class/Hydrator/LayoutDefinitionHydrator.php]: Done here [e0011ca](https://github.com/pimcore/studio-backend-bundle/pull/581/commits/e0011cae0ed4aa168eaaa7a0bd2495b54471abf3)

**Review by lukmzig — APPROVED** [review]
LGTM :boom:

**Pull request merged**